### PR TITLE
Add new RequestMatch authentication method

### PR DESCRIPTION
**Pull request by dbu**
@sagikazarmark found this branch lying around. it seems to add on #34 - is it relevant?

### CODE REVIEW

File: src/Authentication/RequestMatch.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+use Http\Message\Authentication;
+use Http\Message\RequestMatcher;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Authenticate a PSR-7 Request if the request is matching the given request matcher.
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class RequestMatch implements Authentication
```
Comment (dbu) [inline src/Authentication/RequestMatch.php]: maybe we could call this ConditionalAuthentication?

Comment (sagikazarmark) [inline src/Authentication/RequestMatch.php]: We should somehow have the request in the name...otherwise what condition would it be?

**Comment by sagikazarmark** [comment]
Yeah, I replaced the matching auth with the request matcher interface one. It indeed relied on that PR, as the deprecated matcher now uses the callback request matcher internally.

File: src/Authentication/RequestMatch.php (hunk 1 above)
Comment (sagikazarmark) [inline src/Authentication/RequestMatch.php]: RequestConditional?

Comment (sagikazarmark) [inline src/Authentication/RequestMatch.php]: RequestAware?

Comment (sagikazarmark) [inline src/Authentication/RequestMatch.php]: I fear we can't find a self-explanatory name enough, we have to document it anyway.

Comment (dbu) [inline src/Authentication/RequestMatch.php]: i would like RequestConditional better. and explain it - the phpdoc above looks fine to me.

**Pull request merged**